### PR TITLE
Refactor merge

### PR DESCRIPTION
**Pull request by DenverM80**
==28173== LEAK SUMMARY:
==28173==    definitely lost: 0 bytes in 0 blocks
==28173==    indirectly lost: 0 bytes in 0 blocks
==28173==      possibly lost: 0 bytes in 0 blocks
==28173==    still reachable: 2,550 bytes in 16 blocks
==28173==         suppressed: 0 bytes in 0 blocks
==28173== 
==28173== For counts of detected and suppressed errors, rerun with: -v
==28173== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
denverm@dm-lt:~/code/github/ds3_c_sdk_refactor/test$

### CODE REVIEW

File: src/ds3.c [hunk 1]
```diff
@@ -68,25 +57,7 @@ typedef enum {
...
-    size_t (*user_func)(void*, size_t, size_t, void*);
-}ds3_response_data;
-
-typedef struct {
-    ds3_str* key;
-    GPtrArray* values; // A ds3_str list of the header values
-}ds3_response_header;
-
-static void LOG(const ds3_log* log, ds3_log_lvl lvl, const char* message, ...) {
+void LOG(const ds3_log* log, ds3_log_lvl lvl, const char* message, ...) {
```
Comment (rpmoore) [inline src/ds3.c]: Since we are using this in other files we should probably namespace this call

File: src/ds3.c [hunk 2]
```diff
@@ -363,7 +297,7 @@ static size_t _ds3_send_xml_buff(void* buffer, size_t size, size_t nmemb, void*
     return to_read;
 }
 
-static size_t load_buffer(void* buffer, size_t size, size_t nmemb, void* user_data) {
+size_t load_buffer(void* buffer, size_t size, size_t nmemb, void* user_data) {
```
Comment (rpmoore) [inline src/ds3.c]: Since this is used in multiple files we should namespace this call.

File: src/ds3.h [hunk 3]
```diff
@@ -357,6 +356,10 @@ LIBRARY_API ds3_creds*  ds3_create_creds(const char* access_id, const char* secr
 LIBRARY_API ds3_client* ds3_create_client(const char* endpoint, ds3_creds* creds);
 LIBRARY_API ds3_error*  ds3_create_client_from_env(ds3_client** client);
 LIBRARY_API void        ds3_client_register_logging(ds3_client* client, ds3_log_lvl log_lvl, void (* log_callback)(const char* log_message, void* user_data), void* user_data);
+LIBRARY_API void        LOG(const ds3_log* log, ds3_log_lvl lvl, const char* message, ...);
```
Comment (rpmoore) [inline src/ds3.h]: Put this into an internal util header or something like that.  We shouldn't publish this outside of our code.  Same comment for `load_buffer` and `ds3_create_error`

File: src/ds3.c (hunk 1 above)
Comment (DenverM80) [inline src/ds3.c]: ds3_log_message() ?

Comment (rpmoore) [inline src/ds3.c]: Sure

File: src/ds3.c (hunk 2 above)
Comment (DenverM80) [inline src/ds3.c]: ds3_load_buffer()?

Comment (rpmoore) [inline src/ds3.c]: sure

**Pull request closed without merging**